ocaml/xapi/helpers.ml

```diff
@@ -1242,10 +1242,23 @@ end = struct (* can't place these functions in task helpers due to circular depe
   let wait_for_ ~__context ~tasks ~propagate_cancel cb =
     let our_task = Context.get_task_id __context in
     let classes = List.map (fun x -> Printf.sprintf "task/%s" (Ref.string_of x)) (our_task::tasks) in
+    let maybe_cancel_children () =
+      let tasks_str = List.map Ref.short_string_of tasks |> String.concat "; " in
```

Comment:
This can be also be formatted with the Fmt module: `let tasks_str = Fmt.(str "%a" Dump.(list string)) (Ref.short_string_of tasks) in`; it will surround each element with double quotes.